superchain/superchain-version_test.go

```diff
@@ -0,0 +1,61 @@
+package superchain_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/ethereum-optimism/superchain-registry/superchain"
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethclient"
+)
+
+func checkErr(t *testing.T, err error) {
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+// TestContractVersionsCheck will fail if the superchain semver file
+// is not read correctly.
+func TestContractVersionsCheck(t *testing.T) {
+	if err := superchain.SuperchainSemver.SanityCheck(); err != nil {
+		t.Fatal(err)
+	}
+
+	desiredSemver := superchain.SuperchainSemver.OptimismPortal
+	// TODO complete all members of SuperchainSemver
+	// TODO resolve missing contracts
+	// TODO resolve extraneous contracts
+	for _, chain := range superchain.OPChains {
+		client, err := ethclient.Dial(superchain.Superchains[chain.Superchain].Config.L1.PublicRPC)
+		checkErr(t, err)
+		actualSemver, err := getVersion(context.Background(), common.Address(superchain.Addresses[chain.ChainID].OptimismPortalProxy), client) // TODO resolve difference in names
```

Comment:
The desired semver is populated from `semver.yaml`, and gets mapped to `OptimismPortal`. But the actual semver is pulled from a contract with a name with the `Proxy` suffix:
https://github.com/ethereum-optimism/superchain-registry/blob/68a2c8bb2ce15ccaabb051979d383d50d0d332c8/superchain/extra/addresses/goerli/base.json#L8
And similarly for other contracts.

Reply:
I think that makes sense. I was briefly concerned "what if OP Mainnet, which we want to use as our source of truth, doesn't match `semver.yaml`?" But it should never mismatch, and if it does, we'll detect it here in this tests.

Reply:
Agreed. But my real concern here was the mismatch in naming `OptimismPortal` vs `OptimismPortalProxy`. Should these be the same, or not?

Reply:
As per discussion today, contributors are asked to provide proxy addresses. We therefore need to call `version` on those proxy contracts (which will pass the call through to the implementation contract), to read the semantic version we wish to inspect / check. So we will just add the `Proxy` suffix manually in this test.